### PR TITLE
505 feature/tooltip on cron schedules

### PR DESCRIPTION
**Pull request by mariadanieldeepak**
Closes https://github.com/sudar/bulk-delete/issues/505

### CODE REVIEW

**Comment by sudar** [comment]
@mariadanieldeepak 

I am seeing lot of unnecessary changes. Also looks like you have moved the function to a different class.

Did you branched off from the latest `dev/6.0.0` branch before start working on this?

**Comment by mariadanieldeepak** [comment]
@sudar,

> Did you branched off from the latest dev/6.0.0 branch before start working on this?

Yes. I figured out what went wrong. I branched off w/o pulling the latest `dev/6.0.0`. I've fixed this and it should all be fine now.

Also, the issues shown above in Style CI is not because of the files I modified in this PR. I guess we'll have to deal that as a separate issue.

**Comment by sudar** [comment]
@mariadanieldeepak 

Merge latest `dev/6.0.0` to your branch and it should take care of the Style CI issues.

Also show the help message only when the scheduler add-on is installed.

**Comment by mariadanieldeepak** [comment]
@sudar,

I've modified the code to fix the architectural issues that you identified. Screenshot below portrays that the helper text is displayed when Scheduler add-on is activated and hides when it is not.

image

However, not all modules respond to this code change. Let me give you an example.

I've the `Bulk Delete - Scheduler for Deleting Posts by Status Addon` activated (refer screenshot below)

image

But the helper text doesn't appear in the module because the add-on isn't listed under `pro_iterators` variable (`BulkWP.pro_iterators`). And that is because `BD_Scheduler_For_Deleting_Posts_By_Status` class isn't extending `BD_Scheduler_Addon` which adds the add-on slug using `filter_js_array()`

Screenshot: The helper text should be visible, but it is not.

image

Screenshot: `BD_Scheduler_For_Deleting_Posts_By_Status` is not including the slug in `pro_iterators`.

image

Do you want me to modify the Add-on's? (May be split this issue further?)

File: include/Core/Base/Mixin/Renderer.php [hunk 1]
```diff
@@ -408,22 +410,21 @@ protected function render_cron_settings() {
...
+				 *
+				 * @since 6.0.0
+				 *
+				 * @param string                                  $pro_class  HTML class.
+				 * @param string                                  $field_slug Field Slug of module.
+				 * @param \BulkWP\BulkDelete\Core\Base\BaseModule $module     Module.
+				 */
+				apply_filters( 'bd_pro_only_feature_class', $pro_class, $this->field_slug, $this )
+				?>
 				<?php if ( ! empty( $this->scheduler_url ) ) : ?>
```
[the remark this answers is not in the document]
Comment (mariadanieldeepak) [reply include/Core/Base/Mixin/Renderer.php]: Swapping dev between ES6 scoping and PHP's ;) I've reset the changes now.

File: include/Core/Base/Mixin/Renderer.php [hunk 2]
```diff
@@ -432,9 +432,16 @@ protected function render_cron_settings() {
 			</td>
 		</tr>
 
-		<tr>
+		<tr class="<?php if ( ! empty( $pro_class ) ) { echo sanitize_html_class( $pro_class ); } ?>" style="display: none;">
```
Comment (sudar) [inline include/Core/Base/Mixin/Renderer.php]: You can remove the `empty` check and just print it after passing through `sanitize_html_class`.

File: include/Core/Metas/Modules/DeleteUserMetaModule.php [hunk 3]
```diff
@@ -120,8 +120,15 @@ public function render() {
 				</tr>
 
 				<tr>
```
Comment (sudar) [inline include/Core/Metas/Modules/DeleteUserMetaModule.php]: You may have to add the pro class here also right?

**Comment by sudar** [comment]
@mariadanieldeepak 

Create a seperate issue for the add-ons.

**Comment by mariadanieldeepak** [comment]
@sudar,

> Create a seperate issue for the add-ons.

- https://github.com/sudar/bulk-delete-scheduler-for-deleting-posts-by-status/issues/3

File: include/Core/Base/Mixin/Renderer.php (hunk 2 above)
Comment (mariadanieldeepak) [reply include/Core/Base/Mixin/Renderer.php]: > you can remove the `empty` check and just print it after passing through `sanitize_html_class`.

Just realized that `sanitize_html_class` handles empty string. Removed the explicit check.

**Pull request merged**